tsdb/engine/tsm1/compact.go

```diff
@@ -77,10 +76,10 @@ type tsmGeneration struct {
 }
 
 // size returns the total size of the generation
-func (t *tsmGeneration) size() int64 {
-	var n int64
+func (t *tsmGeneration) size() uint32 {
```

Comment:
This should be a `uint64`.  This is the size all all files in the set and not related to the file pointers in any files.  It can exceed a `uint32` if a final compaction run ended up producing a set of files over 4GB (due to rollovers).